Core/TimberApi.Core/ConfiguratorSystem/ConfigurationRepositorySeeder.cs

```diff
@@ -41,9 +41,26 @@ public void AddConfiguratorsFromAssemblyToRepository(Assembly assembly)
             IEnumerable<Type> configuratorTypes = assembly.GetTypesByAttribute<ConfiguratorAttribute>();
             ImmutableArray<IConfigurator> configurators = ValidateAndCreateConfigurators(configuratorTypes).ToImmutableArray();
 
-            _configuratorRepository.AddRange(SceneEntrypoint.MainMenu, GetConfiguratorWithSceneFlag(configurators, SceneEntrypoint.MainMenu));
-            _configuratorRepository.AddRange(SceneEntrypoint.InGame, GetConfiguratorWithSceneFlag(configurators, SceneEntrypoint.InGame));
-            _configuratorRepository.AddRange(SceneEntrypoint.MapEditor, GetConfiguratorWithSceneFlag(configurators, SceneEntrypoint.MapEditor));
+            var filteredConfigurators = FilterConfiguratorsWithoutMissingModDependencies(configurators).ToImmutableArray();
+
+            _configuratorRepository.AddRange(SceneEntrypoint.MainMenu, GetConfiguratorWithSceneFlag(filteredConfigurators, SceneEntrypoint.MainMenu));
+            _configuratorRepository.AddRange(SceneEntrypoint.InGame, GetConfiguratorWithSceneFlag(filteredConfigurators, SceneEntrypoint.InGame));
+            _configuratorRepository.AddRange(SceneEntrypoint.MapEditor, GetConfiguratorWithSceneFlag(filteredConfigurators, SceneEntrypoint.MapEditor));
+        }
+        
+        /// <summary>
+        ///     Configurator filter based on defined mod dependencies
+        /// </summary>
+        /// <param name="configurators">validated configurators</param>
+        /// <returns>Filtered configurators without missing mod dependencies</returns>
+        private IEnumerable<IConfigurator> FilterConfiguratorsWithoutMissingModDependencies(IEnumerable<IConfigurator> configurators)
+        {
+            foreach (var configurator in configurators)
+            {
+                var attribute = configurator.GetType().GetCustomAttribute<RequiredModDependencies>();
+                if(attribute == null || attribute.ModDependencies.All(modDependency => _modRepository.TryGetByUniqueId(modDependency, out _))) 
```

Comment:
Using ` TryGetByUniqueId` doesn't look good. Better to add an `Has` method to the repository to check if a dependency exists.